src/avram/queryable.cr

```diff
@@ -243,6 +243,22 @@ module Avram::Queryable(T)
     end
   end
 
+  # Remove when PG::PGValue matches rs.read possibilities
+  alias PGValue = Bool | Float32 | Float64 | Int32 | Int64 | PG::Numeric | String | Time | UUID | Nil
+
+  def group_count : Hash(Array(PGValue), Int64)
+    database.query_all(
+      query.select_direct(query.groups + ["COUNT(*)"]).statement,
+      args: query.args,
+      queryable: schema_class.name,
+    ) do |rs|
+      {
+        query.groups.map { rs.read PGValue },
+        rs.read Int64,
+      }
```

Comment:
oh sweet! Looks like you were able to clean this up quite a bit.

Reply:
Yes, this method looks great